File: src/main/java/com/github/iamniklas/liocore/procedures/variants/BlinkProcedure.java
package com.github.iamniklas.liocore.procedures.variants;

import com.github.iamniklas.colorspaces.ColorRGB;
import com.github.iamniklas.liocore.led.LEDDataBundle;
import com.github.iamniklas.liocore.led.colorspace.LIOColor;
import com.github.iamniklas.liocore.network.LEDUpdateModel;
import com.github.iamniklas.liocore.procedures.Procedure;

public class BlinkProcedure extends Procedure {

    private LEDDataBundle bundle;

    private LIOColor blinkColor;
    private int frames = 10;
    private int modulo = 2;

    public BlinkProcedure(LEDUpdateModel _ledUpdateModel) {
        super(_ledUpdateModel);
        bundle = _ledUpdateModel.bundle;

        blinkColor = LIOColor.fromRGB(bundle.colorPrimary);
        frames = bundle.duration;
        modulo = bundle.modulo;
        steps = frames;
    }

    @Override
    public void start() {
        super.start();
    }

    @Override
    public void update() {
        step++;
        if(step % modulo == 0) {
            strip.setAllPixels(blinkColor);
        }
        else {
            strip.setAllPixels(LIOColor.fromRGB(ColorRGB.BLACK));
        }

        if(step == steps) {
            strip.setAllPixels(LIOColor.fromRGB(ColorRGB.BLACK));
            finishProcedure();
        }
    }

    @Override
    public void updateLEDUpdateModel(LEDUpdateModel _ledUpdateModel) {

    }
}

File: src/main/java/com/github/iamniklas/liocorekotlin/procedures/BlinkProcedure.kt
package com.github.iamniklas.liocorekotlin.procedures

import com.github.iamniklas.liocorekotlin.led.ColorRGB
import com.github.iamniklas.liocorekotlin.led.LEDDataBundle
import java.awt.Color

class BlinkProcedure(_bundle: com.github.iamniklas.liocorekotlin.led.LEDDataBundle) : com.github.iamniklas.liocorekotlin.procedures.Procedure(_bundle) {
    var mBlinkColor: Color
    var mFrames = 10
    override var mModulo = 2
    override fun start() {}
    public override fun update() {
        mStep++
        if (mStep % mModulo == 0) {
            mStrip!!.setAllPixels(mBlinkColor)
        } else {
            mStrip!!.setAllPixels(com.github.iamniklas.liocorekotlin.led.ColorRGB.black.toSystemColor())
        }
        if (mStep == mSteps) {
            mStrip!!.setAllPixels(com.github.iamniklas.liocorekotlin.led.ColorRGB.black.toSystemColor())
            finishProcedure()
        }
    }

    init {
        mBlinkColor = _bundle.colorPrimary!!.toSystemColor()
        mFrames = _bundle.duration!!
        mModulo = _bundle.modulo!!
        mSteps = mFrames
    }
}